src/uu/cp/src/cp.rs

```diff
@@ -1586,6 +1608,50 @@ fn copy_file(
         return Ok(());
     }
 
+    // issue 5031 case
+    // touch a && ln -s a b && mkdir c
+    // cp --preserve=links -R -H a b c
```

Comment:
I'd like to see some actual explanation here. Just the test case is a bit hard to understand if your unfamiliar with the issue.